webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/services/queries/TestQueryProcessor.scala

```diff
@@ -800,9 +802,13 @@ class TestQueryProcessor extends Loggable {
       """).openOrThrowException("For tests"),
       s(1) :: Nil)
 
-    testQueries(q1 :: q2 :: q3 :: q4 :: q5 ::  q6 :: q7 :: q8 :: Nil, true)
+    val andQueries = q1 :: q2 :: q3 :: q4 :: q5 ::  q6 :: q7 :: q8 :: Nil
+    // And and Or must yield same results when there is only one criteria for node prop, see: #19538
+    val orQueries = andQueries.map(_.modify(_.query.composition).setTo(Or))
+    testQueries(andQueries ::: orQueries, false)
```

Comment:
add the same check for and and or queries on properties (there is only one criteria, it leads to the same results)